Makefile

```
@@ -26,6 +26,41 @@
+# makes the documented gate work the same way everywhere.
+#
+# `?=` leaves an operator-supplied value untouched: make treats environment
+# variables as already set, so an explicit GOROOT still wins.
+GOROOT ?= $(shell go env GOROOT 2>/dev/null)
+GOCACHE ?= $(shell go env GOCACHE 2>/dev/null)
+export GOROOT
+export GOCACHE
+
+# Fail loudly rather than let the sandbox produce confusing import errors
+# several minutes into a run.
+check-goroot:
```

checkmake (0.3.2)

[warning] Target body for "check-goroot" exceeds allowed length of 5 lines (6).

(maxbodylength)